### ISSUE TITLE
would you please provide more detailed instructions for fine-tuning Long-CLIP?

### ISSUE THREAD

**Issue by YiguoHe** [beichenzbc/Long-CLIP#35]
Hello, thank you very much for your outstanding work! 
I would like to ask if you could provide more detailed instructions for fine-tuning that are suitable for beginners. 
As we all know, most laboratories do not have sufficient computational resources, so we are mainly focusing on downstream fine-tuning tasks. For example, due to a lack of adequate hardware resources, we plan to use our own domain-specific long-caption image-text matching dataset to fine-tune your LongCLIP in order to obtain a domain-specific Long-CLIP model. Could you please provide some tutorials on fine-tuning Long-CLIP? For instance, what should the storage format of the dataset be? What are the commands for fine-tuning? Or, which parts of your code should we pay attention to? Thank you very much!

**Comment by zer0int** [comment]
I hope you don't mind me just "jumping in", but I believe my fork of this repo may be relevant to you:
[https://github.com/zer0int/Long-CLIP](https://github.com/zer0int/Long-CLIP)

I have included scripts for dataset preparation. Besides a "normal" script for fine-tuning, I also have scripts for Geometric Parametrization (GmP) fine-tuning. Using GmP, I was able to improve LongCLIP's (and OpenAI/CLIP's) accuracy on a general dataset (CoCo) by fine-tuning it on a single RTX4090 (!). Feel free to check it out - I sure hope it helps you with your project!

**Comment by beichenzbc** [comment]
Thanks for @zer0int , here are some additional points, I hope it will be helpful to you.
For dataset, a dataset with format **(image, long_caption, short_caption)** is plug-and-play to the training code. If you can't obtain suitable short_caption and only have **(image, long_caption)** , you may need to refine "_model.forward()_" in "_model_longclip.py_" to delete the short-text image alignment part.

After data preparation, you may change the dataset in train.py, and change the model loading in train.py from "_load_from_clip_" to "_load_" to further fine-tune our Long-CLIP model.

**Comment by YiguoHe** [comment]
> I hope you don't mind me just "jumping in", but I believe my fork of this repo may be relevant to you: https://github.com/zer0int/Long-CLIP
> 
> I have included scripts for dataset preparation. Besides a "normal" script for fine-tuning, I also have scripts for Geometric Parametrization (GmP) fine-tuning. Using GmP, I was able to improve LongCLIP's (and OpenAI/CLIP's) accuracy on a general dataset (CoCo) by fine-tuning it on a single RTX4090 (!). Feel free to check it out - I sure hope it helps you with your project!

Thank you so much! Your kindness and helpfulness mean a lot to me. Wishing you all the best!

**Comment by YiguoHe** [comment]
> Thanks for @zer0int , here are some additional points, I hope it will be helpful to you. For dataset, a dataset with format **(image, long_caption, short_caption)** is plug-and-play to the training code. If you can't obtain suitable short_caption and only have **(image, long_caption)** , you may need to refine "_model.forward()_" in "_model_longclip.py_" to delete the short-text image alignment part.
> 
> After data preparation, you may change the dataset in train.py, and change the model loading in train.py from "_load_from_clip_" to "_load_" to further fine-tune our Long-CLIP model.

Thank you very much for your explanation! I understand now. The main modification needed is to change the import part from clip to LongCLIP. Then, we need to prepare the data in the format of your pre-trained dataset, or adjust the dataset class definition accordingly. Thanks again!
